### DISCUSSION TITLE
NormalizeError: Requested and resolved page mismatch: /vpn/../vpns/cfg/smb.conf /vpns/cfg/smb.conf

### DISCUSSION THREAD

**Discussion by silentim** [vercel/next.js#36886] [Help]
### Verify canary release

- [ ] I verified that the issue exists in Next.js canary release

### Provide environment information

server: centos 7.9
next: 12.1.5-12.1.6

### What browser are you using? (if relevant)

chrome

### How are you deploying your application? (if relevant)

next build && next start

### Describe the Bug

In my project, I use "GetStaticPaths" and "GetStaticProps" for data fetching. I found some pages go to 404 some times (occasionally, not frequently), and report error like:

NormalizeError: Requested and resolved page mismatch: /vpn/../vpns/cfg/smb.conf /vpns/cfg/smb.conf

![20220513183756](https://user-images.githubusercontent.com/65599013/168269714-211935d6-5483-4e11-b50b-b24c3afb3d7a.png)

### Expected Behavior

no error, and go to normal page

### To Reproduce

next build
next start

**Comment by balazsorban44** [comment]
Hi, could you add a small reproduction?

**Comment by silentim** [comment]
Maybe cannot reproduct, as I said, it occurred occasionally, not always.

Simple codes like:

```
export const getStaticPaths: GetStaticPaths = async () => {
    return {
        paths: [{ params: { postId: '111' }  }],
        fallback: true,
    };
};
```

```
export const getStaticProps: GetStaticProps = async ({ params }) => {
    const { postId } = params;
    if(!postId) return {notFound:true}

    const data = await fetch(`http://someremote.com/api/post?id=${postId}`);
    if(!data) return {notFound:true}

    return { props: { data }, revalidate: 120 }
}
```

**Reply by balazsorban44** [reply]
As a side note, keep in mind that instead of fetching from API routes, you should import the data-fetching logic directly, as `getStaticProps` runs server-side only.

https://nextjs.org/docs/basic-features/data-fetching/get-static-props#write-server-side-code-directly

**Reply by silentim** [reply]
yes, I fetched data from a remote API, here only for an example, means read data there  :(

**Comment by balazsorban44** [comment]
The problem seems to be `../`. This kind of path traversing is not supported by Next.js, and should throw a 400 (in newer versions) or a 500 error.

**Reply by silentim** [reply]
So, should be fixed by Next.js, or another module sponsor?
